From d260a99e2716e2e20b6e0fd8c819e605c4418827 Mon Sep 17 00:00:00 2001
From: Wunk <wunkolo@gmail.com>
Date: Thu, 10 Sep 2026 08:51:58 -0700
Subject: [PATCH] renderer_vulkan: Skip zero-area renderpasses (#2510)

Found in 3D Thunder Blade. The intersection between the viewport and the surface-rect can sometimes not intersect at all, leading to zero-area render-passes. These draws should be culled entirely to avoid creating zero-area render-passes in Vulkan.
---
 src/video_core/renderer_vulkan/vk_rasterizer.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/video_core/renderer_vulkan/vk_rasterizer.cpp b/src/video_core/renderer_vulkan/vk_rasterizer.cpp
index 5dc31c6e9..324fb0606 100644
--- a/src/video_core/renderer_vulkan/vk_rasterizer.cpp
+++ b/src/video_core/renderer_vulkan/vk_rasterizer.cpp
@@ -560,6 +560,11 @@ bool RasterizerVulkan::Draw(bool accelerate, bool is_indexed) {
         return true;
     }
 
+    const auto draw_rect = fb_helper.DrawRect();
+    if (draw_rect.GetHeight() * draw_rect.GetHeight() == 0) {
+        return true;
+    }
+
     pipeline_info.state.attachments.color = framebuffer->Format(SurfaceType::Color);
     pipeline_info.state.attachments.depth = framebuffer->Format(SurfaceType::Depth);
 
@@ -588,7 +593,6 @@ bool RasterizerVulkan::Draw(bool accelerate, bool is_indexed) {
     UploadUniforms(accelerate);
 
     // Begin rendering
-    const auto draw_rect = fb_helper.DrawRect();
     renderpass_cache.BeginRendering(framebuffer, draw_rect);
 
     // Configure viewport and scissor
-- 
2.43.0

